modules/base/tool.go

```diff
@@ -28,6 +29,15 @@ import (
 	"github.com/dustin/go-humanize"
 )
 
+// Use at most this many bytes to determine Content Type.
+const sniffLen = 512
```

Comment:
mostly we use 1024 bytes - I think that's what works best for most external libraries too.

Reply:
would it be possible to add a test to ensure that if a file contains SVG beyond the sniff length that the SVG is still stripped? I could imagine it being something that unfamiliar developers might want to remove and we must not allow it to sneak back in unsafely.

Reply:
I copied this from the [`go http module`](https://github.com/golang/go/blob/dev.boringcrypto.go1.15/src/net/http/sniff.go#L13) that was being used before.  I can add a test, but I'm pretty sure 512 is what was being used before.

Reply:
I stand corrected! I'm misremembering the bytes/runes thing and lfs etc.

Would you be able to add a test that ensures a file with the svg tag after 512 chars still gets sanitized?

Reply:
(I know your code doesn't change this - it's protecting against someone else removing the sanitization later by mistake

Reply:
I'd be happy to add a test, but to be clear, there is no sanitizing or stripping in this PR.  The security is by the nature of how `<img>` tags and content-security-policies work.  So I'm not sure what the test would actually assert.  If the svg tag weren't in the first 512 bytes, the file would get detected as something else, such as text/plain, text/html, or text/xml and served as that.  When viewing the file, gitea would display it as a text file.  When viewing the raw file, the browser will render it as text.  When embedded in an `<img>` tag from a markdown file, the image would fail to display.

Reply:
BTW, as it's currently implemented, we only attempt to look for an SVG content type if it's already been detected as one of: text/plain, text/html, text/xml.  So it's not possible that having the svg tag later in the file than 512 bytes would cause a file to be detected as something other than those three content types.

Reply:
That's exactly my point - we want the HTML file with embedded SVG after 512 to have the SVG sanitised away and displayed without it.

The test would simply assert that there were no SVG els.

The point is to prevent some well meaning change to the sanitizer leading to unsanitised SVG leaking.

Reply:
If you really don't want to do it - it can be added later but it'd be good to get the test in as an example of something that is not going to be supported and also something we don't want to happen.

Reply:
I think there's a misunderstanding.  Maybe I'm not being clear enough.  If there's a file that's detected as text/html with a root `<svg>` tag after 512 bytes, and we serve it with `Content-Type: text/html`, the browser will never up-convert to `image/svg+xml` because we're also  serving the header [`X-Content-Type-Options: nosniff`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options), which tells the browser, "take our word for it; don't try to do your own sniffing".

So I don't see a need to assert that there are no `<svg>` tags.  Am I missing something?

Reply:
I'm working on a test that asserts that a file with a root `<svg>` tag after 512 bytes is detected as `text/html`.  Hopefully, that's what you mean.